pkg/pool-weighted/contracts/managed/vendor/BasePool.sol

```diff
@@ -285,16 +253,15 @@ abstract contract BasePool is
             super._isOwnerOnlyAction(actionId);
     }
 
-    function _getMiscData() internal view returns (bytes32) {
-        return _miscData;
+    function _getPoolState() internal view returns (bytes32) {
+        return _poolState;
     }
 
     /**
-     * @dev Inserts data into the least-significant 192 bits of the misc data storage slot.
-     * Note that the remaining 64 bits are used for the swap fee percentage and cannot be overloaded.
+     * @dev Inserts data into the pool state storage slot.
      */
-    function _setMiscData(bytes32 newData) internal {
-        _miscData = _miscData.insertBits192(newData, 0);
+    function _setPoolState(bytes32 newPoolState) internal {
+        _poolState = newPoolState;
     }
```

Comment:
Is this just to have more explicit state access syntax?

Reply:
Imagine if we had `this.poolState = foo` :smiling_face_with_tear:

Reply:
You could try a simple bytecode size measurement. Reading the slot shouldn't be much more than `push1 slot sload`, whereas the function call requires pushing the destination, jump, jumpdest, etc.